[otbn/sca] Improve OTBN SCA capture setup

vrozic opened the pull request:
This PR contains two updates to improve OTBN ECC256 keygen capture setup:

- Mute ibex during the OTBN operation to reduce noise
- Move seed masking outside of the batch capture loop to prevent "fake leakage".

wettermo commented on sw/device/sca/ecc256_keygen_serial.c:
```diff
@@ -60,6 +60,13 @@ enum {
+   * to complete the operation. Otherwise, Ibex might wake up while OTBN is
+   * still busy and disturb the capture.
+   */
+  kIbexOtbnSleepCycles = 800,
```
Is this an estimated cycle count, or did you measure how many cycles the OTBN takes?

vrozic replied:
For now I've just selected `800` as a placeholder value. It is enough to mute ibex for the 3500 sample measurements that we are currently running.
Eventually, we plan to make this value programmable by the `capture.py` script, but I wouldn't want to do that in this PR.

jadephilipoom commented on sw/device/sca/ecc256_keygen_serial.c:
```diff
@@ -219,24 +226,28 @@ static void ecc256_ecdsa_secret_keygen_batch(const uint8_t *data,
       }
     }
+    for (uint32_t j = 0; j < kEcc256SeedNumWords; ++j) {
+      batch_share0[i][j] = batch_share1[i][j] ^ batch_share0[i][j];
```
nit:
```suggestion
      batch_share0[i][j] ^= batch_share1[i][j]
```

vrozic replied:
Thanks. This is fixed now.

jadephilipoom reviewed (approved):
Seems like this partially addresses https://github.com/lowRISC/opentitan/issues/17338 (partial since there are other files to add sleep functionality to)

abdullahvarici reviewed (approved):
Thanks @vrozic! I also tried it out and it is working. LGTM.

vrozic commented on sw/device/sca/ecc256_keygen_serial.c:
```diff
@@ -297,8 +314,14 @@ static void p256_ecdsa_gen_secret_key(const uint32_t *seed,
                                    uint32_t *d0, uint32_t *d1, uint32_t *x,
                                    uint32_t *y) {
+  // Compute first share of seed (seed ^ mask).
+  uint32_t seed0[kEcc256SeedNumWords];
```
Agreed. It's fixed now

vrozic commented:
Thanks @wettermo , @jadephilipoom and @abdullahvarici for the reviews. I've addressed the comments and added one small fix: Since seed masking was factored out of `p256_run_keygen()` it needed to be added back in `p256_ecdsa_gen_secret_key()` and `p256_ecdsa_gen_keypair()` before calling `p256_run_keygen()`.
